src/write.rs

```diff
@@ -203,7 +203,8 @@ impl<W: Write> BzDecoder<W> {
     ///
     /// [`write`]: Self::write
     pub fn try_finish(&mut self) -> io::Result<()> {
-        while !self.done {
+        // If nothing was written, there is no need to loop
+        while !self.done && self.total_in() > 0 {
```

Comment:
Maybe this should actually return an error when `self.total_in() == 0`? An empty file is not a valid bzip2 file.

Reply:
I don't know, should we return error if we `try_finish` a writer without ever writing anything? I think maybe yes, we should at least pass in the empty bzip2 payload. If we didn't, we have a logic error somewhere on the caller side.

Considering that the code path was "protected" by an infinite loop, returning an error is not a breaking change :smile:

That said, I completely forgot the context in which I made the PR :disappointed:

Reply:
returning an error on an empty input sounds like the best approach to me. The doc comment should reflect that change

Reply:
hmm although doesn't that break the `write_empty` test below? also what happens when you write just part of a bzip file (e.g. the first 2 bytes of the header) and then flush? That behavior should be consistent.